Sources/ProjectDescription/BuildAction.swift

```diff
@@ -20,11 +55,37 @@ public struct BuildAction: Equatable, Codable, Sendable {
     ///   - postActions: A list of actions that are executed after the build process.
     ///   - runPostActionsOnFailure: Whether the post actions should be run in the case of a failure
     /// - Returns: Initialized build action.
+    @available(
+        *,
+        deprecated,
+        message: "Use the initializer that accepts targets as [BuildAction.Target], which supports passing the actions when targets build for each target in the list."
+    )
     public static func buildAction(
         targets: [TargetReference],
         preActions: [ExecutionAction] = [],
         postActions: [ExecutionAction] = [],
         runPostActionsOnFailure: Bool = false
+    ) -> BuildAction {
+        BuildAction(
+            targets: targets.map { Target($0, buildFor: nil) },
+            preActions: preActions,
+            postActions: postActions,
+            runPostActionsOnFailure: runPostActionsOnFailure
+        )
+    }
+
+    /// Returns a build action.
+    /// - Parameters:
+    ///   - targets: A list of targets to build, which are defined in the project.
+    ///   - preActions: A list of actions that are executed before starting the build process.
+    ///   - postActions: A list of actions that are executed after the build process.
+    ///   - runPostActionsOnFailure: Whether the post actions should be run in the case of a failure
+    /// - Returns: Initialized build action.
+    public static func buildAction(
+        _ targets: [BuildAction.Target],
```

Comment:
Note the `_` here to prevent ambiguity when `.buildAction` is invoked from manifests with `[]` as argument.